src/test/ui/lint/liveness-dead.stderr

```diff
@@ -0,0 +1,30 @@
+warning: value assigned to `x` is never read
+  --> $DIR/liveness-dead.rs:21:10
+   |
+LL |     let (mut x, y) = (3, 4); //~ WARN: value assigned to `x` is never read
+   |          ^^^^^       ^^^^^^
```

Comment:
1) The second span isn't correct, as it should be pointing only to `3`, not the entire tuple

2) It doesn't seem to be as useful in this case, I feel that pointing to the reassignment of `x` in line 22 would be better instead:

```
warning: value assigned to `x` is never read
  --> $DIR/liveness-dead.rs:21:10
   |
LL |     let (mut x, y) = (3, 4);
   |          ^^^^^        - this value is never read...
LL |     x = 4;
   |     - ...because its binding is reassigned here before being used

```

3) I would prefer if the second span where a secondary span (blue), instead of both spans being primary spans (red).

Reply:
It's going to be fairly hard (and not always possible) to highlight the value in a case like that. Consider `let (x, y) = pair;` But I do like the second label! Right now I don't think we have that information though. (Grr, we should rewrite liveness in terms of MIR)

Reply:
The language `value from here is never read` rather than `this value`... may make up for our inability to get a variable-specific span on the right hand side. (Ordinarily I'd prefer to avoid "here" language for the same reason it should be avoided in hyperlinks, but "value from assignment" feels less clear?)